drivers/soundwire/slave.c

```diff
@@ -53,6 +53,7 @@ int sdw_slave_add(struct sdw_bus *bus,
 	slave->dev.of_node = of_node_get(to_of_node(fwnode));
 	slave->dev.type = &sdw_slave_type;
 	slave->dev.groups = sdw_slave_status_attr_groups;
+	ACPI_COMPANION_SET(&slave->dev, to_acpi_device_node(fwnode));
```

Comment:
This is basically `set_primary_fwnode(&slave->dev, fwnode);`. No need to use macros.